pennylane/devices/_legacy_device.py

```diff
@@ -480,7 +480,7 @@ def execute(self, queue, observables, parameters=None, **kwargs):
                 elif isinstance(mp, StateMP):
                     raise qml.QuantumFunctionError("Returning the state is not supported")
 
-                elif mp._shortname is not None: # pylint: disable=protected-access
+                elif mp._shortname is not None:  # pylint: disable=protected-access
```

Comment:
```suggestion
                else:
```
Yes, this code is a graveyard, but this seems like a better workaround. We shouldn't just silently ignore measuerment processes we haven't seen.

Reply:
@albi3ro I tried this initially and noticed some tests indeed failed due to this. Couldn't locate it rn but let's try apply and retrigger to collect

Reply:
@albi3ro `tests/devices/test_qubit_device.py::TestExtractStatistics::test_results_created_empty` and its sibling in qutrit. https://github.com/PennyLaneAI/pennylane/actions/runs/12933400215/job/36071950367?pr=6841

They seem to expect an empty return, quite suspicious but in case any other consideration I did not catch so I just left the conditional branch as before.

Reply:
Basically, these tests are examples of testing problematic behaviour that nobody actually wants.